apps/aecore/src/aec_tx_pool.erl

```diff
@@ -427,10 +436,11 @@ fold_txs([], Gas, _, _, _, _, _, _, Acc) ->
     {Gas, Acc}.
 
 int_get_candidate_({?KEY(_, _, Account, Nonce, _) = Key, Tx},
-                   Gas, MinMinerGasPrice, Db, Dbs, AccountsTree, Height, Acc) ->
-    case gb_trees:is_defined({Account, Nonce}, Acc) of
-        true ->
-            %% The earlier must have had higher fee. Skip this tx.
+                   Gas, MinMinerGasPrice, Db, Dbs, AccountsTree, Height,
+                   Acc = #{ tree := AccTree }) ->
+    case gb_trees:is_defined({Account, Nonce}, AccTree) of
+        true when Nonce > 0 ->
```

Comment:
So this does not apply for meta txs (having a [nonce = 0](https://github.com/aeternity/aeternity/blob/master/apps/aega/src/aega_meta_tx.erl#L155-L157)